src/js/tech/tech.js

```diff
@@ -218,12 +218,13 @@ class Tech extends Component {
       return;
     }
 
+    let player = this.player();
     let textTracksChanges = function() {
       let updateDisplay = Fn.bind(this, function() {
-        this.trigger('texttrackchange');
+        player.trigger('texttrackchange');
```

Comment:
The main issue was that the display's listener was calling the wrong method. Though, when I changed that, it seemed to not work still.
Also, I'm actually not a fan of the `texttrackchange` event since it isn't in the spec. Though, perhaps there isn't a better way of doing it.



> > I might keep the listener in there and then have the tech tell the display to update its listeners.

> If there's a way we can do that through an event instead of a direct connection between the tech and the display, that'd be ideal.

If there was an event for tech switching, the display could listen to it :)